docs/proposals/horovod-on-tony.md

```diff
@@ -0,0 +1,113 @@
+- [Motivation](#motivation)
+- [Goals](#goals)
+- [API](#api)
+- [Design](#design)
+- [Usage](#usage)
+
+## Motivation
+The purpose of Horovod is to make it easy to take a single-GPU training script and successfully scale it to train across many GPUs in parallel. This has two aspects:
+1. Easy to use
+2. Run faster in distributed mode
+
+However, because of the limitation of SSH mechanism (hard to support SSH on Yarn), TonY don't support horovod with **MPI controller**. With the help of **Gloo controller**, we are expecting to support horovod.
+
+## Goals
+1. Support Horovod on TonY (limited static topology support)
+2. Elastic Horovod will be supported later.
+
+## API
+There are no API changes to other machine learning frameworks.
+
+## Design
+Attention: It's gloo controller that makes TonY support horovod.
+
+From the perspective of compatibility and maintainability, it's better to directly use the [gloo_runner](https://github.com/horovod/horovod/blob/master/horovod/runner/gloo_run.py) on TonY. But after reading through Horovod's code, I find it difficult to reuse gloo_runner's code on TonY, because the existence of some unrelated codes will lead driver to start the worker through the SSH command, which is hard to be supportted on Yarn.
```

Comment:
Will there be any compatibility issue later if Horovod makes changes?

Reply:
Yes. 
If horovod changes the core api about rendezvous server, TonY needs to be compatible  with it.